src/core/lib/iomgr/exec_ctx.h

```diff
@@ -176,30 +176,26 @@ class ExecCtx {
     }
   }
 
-  /** Returns the stored current time relative to start if valid,
-   *  otherwise refreshes the stored time, sets it valid and returns the new
-   *  value.
+  /** Returns the current time relative to start.
+   *  TODO(yashykt): Remove and replace with something not dependent on ExecCtx.
    */
   Timestamp Now();
 
-  /** Invalidates the stored time value. A new time value will be set on calling
-   *  Now().
+  /** Deprecated.
+   *  TODO(yashykt): Remove all instances of this.
```

Comment:
Any reason not to do this as part of this PR?

Reply:
I wanted to make sure this doesn't break anything before spending too much time on this